vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
A few comments.

1) Is there a reason for setting _message to None?

2) I hadn't thought much about the open transaction issue. Now, I support @jberka 's previous comment about not supporting open transactions. Supporting this can lead to unexpected behavior. An example would be: user runs a SELECT query --> iterate/process the results and break the loop in the middle (a reason could be the condition is too complex to be embedded in the query) --> he runs another query --> undefined behavior because there are data left from the earlier query.